packages/salesforcedx-vscode-apex/src/languageServer.ts

```diff
@@ -29,77 +25,43 @@ const DEBUG = typeof v8debug === 'object' || startedInDebugMode();
 
 async function createServer(
   context: vscode.ExtensionContext
-): Promise<StreamInfo> {
+): Promise<Executable> {
   try {
     deleteDbIfExists();
     const requirementsData = await requirements.resolveRequirements();
-    return new Promise<any>((resolve, reject) => {
-      portFinder.getPort((err, port) => {
-        const uberJar = path.resolve(
-          context.extensionPath,
-          'out',
-          UBER_JAR_NAME
-        );
-        const javaExecutable = path.resolve(
-          `${requirementsData.java_home}/bin/java`
-        );
-        let args: string[];
-        if (DEBUG) {
-          args = [
-            '-cp',
-            uberJar,
-            `-Dapex-lsp.port=${port}`,
-            '-Ddebug.internal.errors=true',
-            '-Ddebug.semantic.errors=false',
-            '-Dtrace.protocol=false',
-            `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${JDWP_DEBUG_PORT}`,
-            APEX_LANGUAGE_SERVER_MAIN
-          ];
-        } else {
-          args = [
-            '-cp',
-            uberJar,
-            `-Dapex-lsp.port=${port}`,
-            '-Ddebug.internal.errors=true',
-            '-Ddebug.semantic.errors=false',
-            APEX_LANGUAGE_SERVER_MAIN
-          ];
-        }
-
-        net
-          .createServer(socket => {
-            resolve({
-              reader: socket,
-              writer: socket
-            });
-          })
-          .listen(port, () => {
-            const options = {
-              cwd: vscode.workspace.rootPath
-            };
-
-            const lspProcess = child_process.spawn(
-              javaExecutable,
-              args,
-              options
-            );
-            console.log('PROCESS INFO');
-            console.log(lspProcess);
+    const uberJar = path.resolve(context.extensionPath, 'out', UBER_JAR_NAME);
+    const javaExecutable = path.resolve(
+      `${requirementsData.java_home}/bin/java`
+    );
+    let args: string[];
+    if (DEBUG) {
+      args = [
+        '-cp',
+        uberJar,
+        '-Ddebug.internal.errors=true',
+        '-Ddebug.semantic.errors=false',
+        '-Dtrace.protocol=false',
+        `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${JDWP_DEBUG_PORT},quiet=y`,
```

Comment:
Care must be exercised to not put *anything* that is not a JSON/RPC message on the stdout and stdin.
This is why we need the quiet=y flag on the agentlib as well.